script.js

```diff
@@ -1,117 +1,7 @@
-selectEpisode.className = "select xl-col-2 lg-col-2 md-col-3 sm-col-4 col-10";
-searchText.className =
-  "searchInText xl-col-2 lg-col-2 md-col-3 sm-col-3 col-10";
-searchState.className =
-  "searchState xl-col-4 lg-col-5 md-col-6 sm-col-7 col-12";
-
-rootElem.appendChild(searchDiv);
-rootElem.appendChild(episodeDiv);
-
-searchDiv.appendChild(selectShow);
-searchDiv.appendChild(selectEpisode);
-searchDiv.appendChild(searchText);
-searchDiv.appendChild(searchState);
-selectEpisode.style.display = "none";
-searchState.style.display = "none";
-
+//You can edit ALL of the code here
 function setup() {
-  let allShows = getAllShows();
-  console.log(allShows);
-  sortOn(allShows, "name");
-  makePageForShows(allShows);
-  addShow(allShows);
-  searchShows(allShows);
-}
-
-function sortOn(shows, name) {
-  //Alphabeticaly and case insensitive sort the shows by name.
-  shows.sort(function (a, b) {
-    return a[name].localeCompare(b[name]);
-  });
-  return shows;
-}
-
-function makePageForShows(showList) {
-  episodeDiv.innerHTML = "";
-  showList.forEach((show) => {
-    //Iterate through the array of episode.
-    let summaryText = show.summary.replace(/<p>/g, " "); //remove the '<p>' from the summary.
-    let pEl = summaryText.replace(/<\/p>/g, " "); //remove the '</p>' from the summary.
-
-    //Create the elements by innerHTML and add to episodeDiv div.
-    episodeDiv.innerHTML += `
-    <div class="exDiv col-12 sm-col-12 md-col-12 lg-col-12 xl-col-12">
-      <div class="divElStyle2 col-11 sm-col-11 md-col-11 lg-col-11 xl-col-11">
-        <h2 class="h2ElStyle col-12" onclick="fetchEpisodes(${show.id})" id=${show.id}>${show.name}</h2>
-        <div class="detailsShow col-12 sm-col-12 md-col-12 lg-col-12 xl-col-12">
-          <div>
-            <p class="zoomText">Genres: ${show.genres}</p>
-            <p>Status: ${show.status}</p>
-            <p>Rating: ${show.rating.average}</p>
-            <p>Runtime: ${show.runtime}</p>
-            <p>Language: ${show.language}</p>
-          </div>
-          <img src=${show.image.medium}  img>
-        </div>
-        <p class="pElStyle col-11">${pEl}</p>
-      </div>
-    </div>`;
-  });
-}
-
-function searchShows(allShow) {
-  searchText.addEventListener("input", () => {
-    let addSearchedShows = allShow.filter((show) =>
-      show.name.indexOf(searchText.value) > -1 ||
-      show.summary.indexOf(searchText.value) > -1 ? true : false);
-    addShow(addSearchedShows);
-    makePageForShows(addSearchedShows);
-  });
-}
-
-function addShow(showList) {
-  //Add the names and ids of the shows from show list to select show.
-  selectShow.innerHTML = `<option>All shows</option>`;
-  showList.forEach((show) => {
-    selectShow.innerHTML += `<option value="${show.id}">${show.name}</option>`;
-  });
-  selectShow.addEventListener("change", (changeEvent) =>
-    fetchEpisodes(changeEvent.target.value)
-  );
-}
-
-function fetchEpisodes(showId) {
-  selectEpisode.style.display = "block";
```

Comment:
You can move it to CSS, to leave JS for JS.

Reply:
Great, I got the point.